### PR TITLE
feat: implement xlinectl `snapshot` and `auth` commands

### PR DESCRIPTION
**Pull request by bsbds**
Based on: #424

Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)

* what changes does this pull request make?

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)

### CODE REVIEW

File: xlinectl/src/command/snapshot.rs [hunk 1]
```diff
@@ -0,0 +1,44 @@
+use std::fs::File;
+use std::io::Write;
+use std::path::PathBuf;
```
Comment (Phoenix500526) [inline xlinectl/src/command/snapshot.rs:3]: Please merge imports

**Review by themanforfree — DISMISSED** [review]
LGTM

File: xlinectl/src/command/snapshot.rs (hunk 1 above)
Comment (bsbds) [reply xlinectl/src/command/snapshot.rs]: done

**Review by Phoenix500526 — APPROVED** [review]
LGTM

**Review by themanforfree — APPROVED** [review]

**Pull request merged**